.github/workflows/codecov.yml

```diff
@@ -0,0 +1,134 @@
+name: Code Coverage
+on: # only on merges to main
+  push:
+    branches:
+      - main
+concurrency:
+  group: format('{0}-{1}', ${{ github.ref }}, 'Code Coverage')
+  cancel-in-progress: true
+
+permissions: read-all
+
+env:
+  GITHUB_PR_HEAD_SHA: "${{ github.event.pull_request.head.sha }}"
+
+jobs:
+  test:
+    name: Code Coverage
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Skip workflow if it's not a merge to main
+      id: skip-workflow
+      run: |
+        skip='true'
+        if [[ "${{github.event.pull_request}}" ==  "" ]] && [[ "${{github.ref}}" == "refs/heads/main" ]]; then
+          skip='false'
+        fi
+        echo Skip ${skip}
+        echo "skip-workflow=${skip}" >> $GITHUB_OUTPUT
```

Comment:
I think this entire step can be omitted as we run the workflow only on new pushes to main, it is only going to be a merge to main that triggers it.

Reply:
Yeah, related to the comment above -- I'd end up removing this if we decide that we only care about `main` as it becomes redundant.